src/extensions/default/InlineBoxModelEditor/BoxModelEditor.js

```diff
@@ -66,17 +66,31 @@ define(function(require, exports, module) {
         });
     }
 
-    PaddingValue.prototype.toString = function() {
+    BoxModelValue.prototype.toString = function() {
         return this.value + (this.value === 0 ? "" : this.unit);
     };
 
-    function PaddingEditor($parent, valueString, paddingChangeHandler) {
+    function BoxModelEditor($parent, valueString, BoxModelChangeHandler, type, iconClassName) {
         var self = this;
 
+        var headerKey = "";
+
+        switch (type) {
+            case BoxModelUtils.PADDING:
```

Comment:
@bansalnitish I don't think this is right still...
You need to swap the `case` check not the entire `case` block. Out here you're checking if the type is `PADDING` and setting the `headerKey` to `SET_MARGIN_FOR` which is the opposite of what it should do.